proc/test.yaml

```diff
@@ -0,0 +1,7 @@
+test:
+    tests:
+        - name: test_thread_rand
+          exec: test_thread_rand
+          type: unit
+          targets:
+              include: [armv7m7-imxrt106x]
```

Comment:
Marek and me asked you to compile this test on all targets. When this issue will be addressed?

Reply:
Test is now build and tested on all targets that our runner supports (ia32 and imxrt106x). When the automatic tests will be available for imxrt106x (soon) even that include will not be necessary. The same goes with other automatic tests currently available.

The term `include:` does not specify _the only_ architecture but it adds it. Even if I specify all architectures here, tests do not build everywhere - only on imxrt106x, ia32 and host-pc

Reply:
@pawelpisarczyk this is taken care of, `targets` is a dict which supports attributes (`include`, `exclude` and `values`). The default value for `values` is `DEFAULT_TARGETS` (currently `ia32-generic` as all tests should work on this platform).

The final set of targets is `targets['values'] + targets['include'] - targets['exclude']` (yeah, seems unnaturally complex, but common pattern in YAML configs) - works great when You have multiple tests in single YAML and You globally set targets for all of them and then just include/exclude one or two for a certain target)